### PR TITLE
Cleanup composer.json

### PR DESCRIPTION
**Pull request by lucasmichot**
* Sort keywords, require and require-dev
* Suffix folders

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
:-1:

**Comment by lucasmichot** [comment]
Comment?

**Comment by tymondesigns** [comment]
@vinkla added the double pipes in #373 see [here](https://getcomposer.org/doc/articles/versions.md#range)

What's the difference between `5.1.* || 5.2.*` and `~5.1.0|~5.2.0` ?

File: composer.json [hunk 1]
```diff
@@ -20,28 +20,28 @@
     ],
     "require": {
         "php": ">=5.5.9",
-        "illuminate/support": "5.1.* || 5.2.*",
-        "illuminate/http": "5.1.* || 5.2.*",
-        "illuminate/auth": "5.1.* || 5.2.*",
-        "illuminate/contracts": "5.1.* || 5.2.*",
-        "namshi/jose": "6.0.*",
-        "nesbot/carbon": "^1.0"
+        "illuminate/auth": "~5.1.0|~5.2.0",
```
Comment (vinkla) [inline composer.json]: This is dangerous since we don't know if it will break in future versions of Laravel.

**Comment by vinkla** [comment]
Yeah, I'm also :-1: on this.

**Comment by vinkla** [comment]
`5.1.* || 5.2.*` will allow only those specific versions of Laravel. `||` is now recommended by Composer instead of one `|` as we did before.

File: composer.json (hunk 1 above)
Comment (lucasmichot) [inline composer.json]: How can this be dangerous ? We will not reach `5.3.*`

Comment (crynobone) [inline composer.json]: It's not going to install 5.3 when it's available with or without the changes.

**Comment by vinkla** [comment]
I agree we can suffix the folder paths :+1: .

File: composer.json (hunk 1 above)
Comment (lucasmichot) [inline composer.json]: @vinkla see: http://blog.madewithlove.be/post/tilde-and-caret-constraints/

Comment (vinkla) [inline composer.json]: Yeah, I read it wrong. Sorry, removed the comment.

**Comment by tymondesigns** [comment]
So I guess we can leave the version declarations as is, since there is no difference either way: 

5.1._whatever_ OR 5.2._whatever_

happy with everything else

**Comment by lucasmichot** [comment]
@vinkla the `||` notation seems specific to the PHP version, see https://getcomposer.org/doc/04-schema.md#require
I cannot find any assertion of what you say about the `||` notation either in the composer docs, either in https://github.com/composer/composer issues or PRs

The only thing that I see is that a support is offered to npm-like notation in an alpha release : https://github.com/composer/composer/releases/tag/1.0.0-alpha10 

Do you have more infos ?

**Comment by GrahamCampbell** [comment]
:-1:

**Comment by GrahamCampbell** [comment]
If it's not broke, don't change it.

**Comment by vinkla** [comment]
> Yeah `||` is the new cool, npm compatible and the only thing we recommend in docs https://getcomposer.org/doc/articles/versions.md#range
https://github.com/composer/semver/pull/22#issuecomment-151858703

**Comment by lucasmichot** [comment]
> If it's not broke, don't change it.

@GrahamCampbell 

* Sorting keywords and require allow to avoid noise when adding or removing keywords and/or requires
* adding the / suffix allows composer not to do it

Also making things more readable, organized or usage-consitant are not useless

**Pull request merged**